### ISSUE TITLE
processing data option in beforeCreate hook makes it impossible to use this in data option

### ISSUE THREAD

**Issue by beeplin** [vuejs/vue-rx#12]
This plugin runs the `data` function in the `beforeCreate` hook to transform observables to subscriptions, which makes it impossible to use `this` in `data`, since `this` does not exist when `beforeCreate`. 

For example, sometimes we do this: 

```js
new Vue {
  data: function () {
    return {
      route: this.$route.path; // to get the current router path when this vue component is created.
      ...
    }
  }
}
```

But with this plugin we have to change it into this: 

```js
new Vue {
  data: function () {
    return {
      route: ''
      ...
    }
  },
  created: function () {
    this.route = this.$route.path; 
  }
}
```

It's okay for our own codes, but it gets painful when some third-party packages have this issue. In fact many plugins such as `mint-ui` have this issue, since using `this` in `data` is not very rare in practice.

So my suggestion is to move this `data` processing code to `created` hook instead of `beforeCreate` hook. I understand that, in `created`, if we add new properties to `data` we have to manually make them reactive, but in `beforeCreate` we don't need to do this. I guess this is why currently it's in `beforeCreate`. But considering the issue above, I think it's worth to make this change.

**Comment by zoomclub** [comment]
I'm not sure why RxJS has to be integrated in such a cloaked way by Vue (is there some benefit I'm not aware of)? RxJS is what it is, just another API, a utility that can be imported and used where it and it's wonderful operators might find a usage.

Integrating RxJS as a Vue Directive would be better. Then in the directives handler you can use RxJS to it's maximum benefit. I just saw the following awesome article where the Pointer-Events API was integrated into Vue via a Directive. The same integration could be done for RxJS.

https://benjaminlistwon.com/blog/strategies-for-extending-vue

The RxJS API should be just like Lodash or any other API, just an option one is free to use in component method, computed or watch functions.

**Comment by beeplin** [comment]
@zoomclub RxJS may be used to handle either DOM events or back-end socket.io data push. If you are using Vue, I don't think RxJS is quite useful for DOM handling, since Vue (as a **view** tool) already has plenty of gadgets to interact with the DOM. But Vue doesn't provide any **model** or back-end data tools. That's where RxJS can help. I usually use this plugin to link socket.io data push from back-end and vue in the front-end.

**Comment by zoomclub** [comment]
@beeplin Still need a way to turn DOM events into a stream/observable in Vue. A directive would essentially just wrap the following RxJS operator: 

http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-fromEvent

This would be nice:

<div v-rxjs:event-type=“handler”>

**Comment by yyx990803** [comment]
I think it might be better if we change the API to use a separate option instead:

``` js
data () {
  // normal state
  return { ... }
},
rx: {
  key: someObservable$
},
```

This more explicitly separates observables vs. normal state, and solves the issue you are running into.

**Comment by beeplin** [comment]
@yyx990803 sure, that would be great. :)

**Comment by yyx990803** [comment]
released 2.0.0.
